src/index.tsx

```diff
@@ -175,34 +174,25 @@ apiConnection.onMessage(msg => {
     console.warn(`Unregognized message type from server: ${type0}`)
   }
 });
-// setApiConnection(apiConnection);
-const waitForWorkspaceInfo = async () => {
-  while (true) {
-    const state = theStore.getState() as RootState
-    const workspaceInfo = state.workspaceInfo
-    if ((workspaceInfo) && (workspaceInfo.workspaceName)) {
-      apiConnection.sendMessage({
-        type: 'reportClientInfo',
-        clientInfo: {
-          feedUri: workspaceInfo.feedUri,
-          workspaceName: workspaceInfo.workspaceName,
-          readOnly: workspaceInfo.readOnly
-        }
-      })
-      return;
+
+const handleSetWorkspaceInfo = (workspaceInfo: WorkspaceInfo) => {
+  apiConnection.sendMessage({
```

Comment:
This method of reporting client info to the server is way better than previous awkward method of waiting for the redux state to be updated properly.